pycqed/measurement/waveform_control/circuit_builder.py

```diff
@@ -859,8 +859,6 @@ def sweep_n_dim(self, sweep_points, body_block=None, body_block_func=None,
                     sweep_dicts_list=sweep_points, sweep_index_list=[j, i]))
                 # apply Segment sweep points
                 for dim in [0, 1]:
-                    if len(sweep_points[dim]) == 0:
```

Comment:
Side note about wrapper functions: This "if" had been necessary before because the wrapper function return an error when asking for 'all' and there are no sweep points in the dim. By directly accessing the keys, I get an empty iterable if there are no sweep points in the dim. This might also be a reasonable behavior for the wrapper functions. @stephlazar

Reply:
Yes probably. But maybe not part of this PR?

Reply:
No, not part of this PR 👍